explorer/interpreter/interpreter.cpp

```diff
@@ -966,9 +966,13 @@ auto Interpreter::CallFunction(const CallExpression& call,
       return todo_.FinishAction(arena_->New<AlternativeValue>(
           &alt.choice(), &alt.alternative(), cast<TupleValue>(arg)));
     }
-    case Value::Kind::FunctionValue: {
-      const auto& fun_val = cast<FunctionValue>(*fun);
-      const FunctionDeclaration& function = fun_val.declaration();
+    case Value::Kind::FunctionValue:
+    case Value::Kind::BoundMethodValue: {
+      const auto* func_val = dyn_cast<FunctionValue>(fun);
+      const auto* method_val = dyn_cast<BoundMethodValue>(fun);
```

Comment:
Might it be worth adding a shared base class for these, containing declaration, type_args, and witnesses?